### PR TITLE
Added documentation on Cutting a Rod problem

### LINKED ISSUE #434: Article on Rod cutting problem
## 📚 Documentation

Article on Rod cutting problem.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

Please assign this to me under GSSOC'21.

### PR DESCRIPTION
**Pull request by nidh-ish**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added documentation on Cutting a Rod problem.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #434

### CODE REVIEW

File: docs/dynamic-programming/cutting-a-rod.mdx [hunk 1]
```diff
@@ -0,0 +1,178 @@
+---
+title: Cutting a rod
+slug: /dynamic-programming/cutting-a-rod
+---
+
+## Introduction
+
+Given a rod of length `n` and an array of prices of all pieces of length `1` to `n`, we need to determine the maximum price that can be obtained by cutting up the rod and selling its pieces.
```
Comment (HarshCasper) [inline docs/dynamic-programming/cutting-a-rod.mdx:8]: The introduction is not very intuitive. A better explanation would be needed here.

File: docs/dynamic-programming/cutting-a-rod.mdx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+  - Selling price = 5 + 7 = 12
+
+3) Length 5
+
+  - Selling price = 10
+
+Here, we see that a maximum selling price of 12 is generated when the given rod is cut into
+sub-rods of length 2 and 3.
+```
```
Comment (HarshCasper) [inline docs/dynamic-programming/cutting-a-rod.mdx:37]: A diagrammatic representation would be better here

File: docs/dynamic-programming/cutting-a-rod.mdx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+- Initialize an array `dp[]` such that the `ith` element stores the maximum price obtained by a rod of length `i`.
+
+```
+for i = 1 to n
+    current_max = i
+    for j = 0 to i - 1
+        current_max = max(cost[j] + dp[ i- j - 1], current_max)
+    dp[i] = current_max
+return dp[n]
+```
```
Comment (HarshCasper) [inline docs/dynamic-programming/cutting-a-rod.mdx:50]: Don't substitute algorithm steps with pseudocode. There is a reason why we have actual code implementations here. Use Algorithm to strictly describe how your code works step-wise

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by nidh-ish** [comment]
@HarshCasper I have made the changes. Please have a look.

**Comment by nidh-ish** [comment]
@siddharth25pandey @iamrajiv @HarshCasper Please review this PR.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**